scripts/app.js

```diff
@@ -0,0 +1,345 @@
+function getRepoName(url) {
+  let repoName = url.split('/');
+  repoName = repoName[repoName.length - 1];
+  repoName = repoName.replace('.git', '');
+  repoName = repoName.replace('\n', '');
+  return repoName;
+}
+
+//delete patch file
+
+async function deletePatchFile() {
+  return new Promise((resolve, reject) => {
+    exec('rm myPatch.patch', (error) => {
+      if (error !== null) {
+        console.log('exec error deleting patch file: ', error);
+        reject(error);
+      }
+      resolve();
+    });
+  })
+}
+async function getRepoInfo() {
+  return new Promise((resolve, reject) => {
+    exec('git config --get remote.origin.url', (error, stdout) => {
+      if (error !== null) {
+        console.log(`exec error: ${error}`);
+        reject(error);
+      }
+
+      const repoUrl = stdout.replace('\n', '');
+      resolve(repoUrl);
+    });
+  });
+}
+
+async function getGitEmail() {
+  return new Promise((resolve, reject) => {
+    exec('git config --global user.email', (error, stdout) => {
+      if (error !== null) {
+        console.log(`exec error: ${error}`);
+        reject(error);
+      } else {
+        resolve(stdout.replace('\n', ''));
+      }
+    });
+  });
+}
+
+async function getGitUser() {
+  return new Promise((resolve, reject) => {
+    exec('git config --global user.name', (error, stdout) => {
+      if (error !== null) {
+        console.log(`exec error: ${error}`);
+        reject(error);
+      } else {
+        resolve(stdout.replace('\n', ''));
+      }
+    });
+  });
+}
+
+async function getGitCommits() {
+  return new Promise((resolve, reject) => {
+    exec('git cherry', (error, stdout) => {
+      if (error !== null) {
+        console.log(`exec error: ${error}`);
+        reject(error);
+      } else {
+        const cleanedup = stdout.replace(/\+ /g, '');
+        const commitarray = cleanedup.split(/\r\n|\r|\n/);
+        commitarray.pop(); //remove the last, dummy element that results from splitting on newline
+        if (commitarray.length === 0) {
+          console.log('You have tried to create a staging job from local commits but you have no committed work. Please make commits and then try again');
+          reject();
+        }
+        if (commitarray.length === 1) {
+          const firstCommit = commitarray[0];
+          const lastCommit = null;
+          resolve({ firstCommit, lastCommit });
+        } else {
+          const firstCommit = commitarray[0];
+          const lastCommit = commitarray[commitarray.length - 1];
+          resolve({ firstCommit, lastCommit });
+        }
+      }
+    });
+  });
+}
+async function getGitPatchFromLocal(branchName) {
+  return new Promise((resolve, reject) => {
+    exec(`git diff origin/${branchName} > myPatch.patch`, (error) => {
+      if (error !== null) {
+        console.log('error generating patch: ', error);
+        reject(error);
+      } else {
+        fs.readFile('myPatch.patch', 'utf8', (err, data) => {
```

Comment:
Not clear on the requirement here -- we want to run a patch build even if the patch creation fails?